jetty-jmx/src/main/java/org/eclipse/jetty/jmx/MBeanContainer.java

```diff
@@ -242,7 +242,7 @@ private static MetaData newMetaData(MBeanContainer container, Class<?> klass)
             return null;
         String pName = pkg.getName();
         String cName = klass.getName().substring(pName.isEmpty() ? 0 : pName.length() + 1);
-        String mName = pName + ".jmx." + cName + "MBean";
+        String mName = pName + ".jmx." + cName.replaceAll("\\$", "I") + "MBean";
```

Comment:
Can't have $ in the classname unless it really is an inner class.

One solution is to just not have so many inner classes.... but I'm finding it useful to have less files to look at whilst doing the refacting.    But ultimately we need to review if we should move many of those inner classes to the top level.

However, either way it would be good if the MBean mechanism could be made to work for inner classes.  

Would "Inner" work better?   So if we are looking for a MBean for `Foo$Bar` we would look for `FooInnerBarMBean`?
Or maybe flip, so we look for `BarInFooMBean` ?